diff --git a/node_modules/@nozbe/watermelondb/decorators/experimentalFailsafe/index.d.ts b/node_modules/@nozbe/watermelondb/decorators/experimentalFailsafe/index.d.ts
new file mode 100644
index 0000000..6da29e6
--- /dev/null
+++ b/node_modules/@nozbe/watermelondb/decorators/experimentalFailsafe/index.d.ts
@@ -0,0 +1,2 @@
+declare function failsafe(fallback?: any): PropertyDecorator
+export default failsafe
\ No newline at end of file
diff --git a/node_modules/@nozbe/watermelondb/decorators/index.d.ts b/node_modules/@nozbe/watermelondb/decorators/index.d.ts
index 7f1fc8e..9fb334a 100644
--- a/node_modules/@nozbe/watermelondb/decorators/index.d.ts
+++ b/node_modules/@nozbe/watermelondb/decorators/index.d.ts
@@ -9,3 +9,4 @@ export { default as readonly } from './readonly'
 export { default as lazy } from './lazy'
 export { default as relation } from './relation'
 export { default as immutableRelation } from './immutableRelation'
+export { default as experimentalFailsafe } from './experimentalFailsafe'
