diff --git a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
index 32a24f3..f50a67c 100644
--- a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
+++ b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
@@ -146,8 +146,13 @@ public class ReanimatedNativeHierarchyManager extends ReanimatedNativeHierarchyM
       @Nullable ViewAtIndex[] viewsToAdd,
       @Nullable int[] tagsToDelete) {
     if (isLayoutAnimationDisabled()) {
-      super.manageChildren(tag, indicesToRemove, viewsToAdd, tagsToDelete);
-      return;
+      try {
+        super.manageChildren(tag, indicesToRemove, viewsToAdd, tagsToDelete);
+        return;
+      } catch (Exception e) {
+        e.printStackTrace();
+        return;
+      }
     }
     ViewGroup viewGroup;
     ViewGroupManager viewGroupManager;
diff --git a/node_modules/react-native-reanimated/src/mock.ts b/node_modules/react-native-reanimated/src/mock.ts
index a6ff0ce..40c0192 100644
--- a/node_modules/react-native-reanimated/src/mock.ts
+++ b/node_modules/react-native-reanimated/src/mock.ts
@@ -31,6 +31,7 @@ import {
   SensorType,
   setUpTests,
   SharedTransitionType,
+  useReducedMotion,
   withReanimatedTimer,
 } from './index';
 
@@ -119,6 +120,7 @@ const hook = {
   }),
   // useFrameCallback: ADD ME IF NEEDED
   useAnimatedKeyboard: () => ({ height: 0, state: 0 }),
+  useReducedMotion: () => false,
   // useScrollViewOffset: ADD ME IF NEEDED
 };
 
