diff --git a/node_modules/react-native-app-auth/index.d.ts b/node_modules/react-native-app-auth/index.d.ts
index 0c59626..2debfdb 100644
--- a/node_modules/react-native-app-auth/index.d.ts
+++ b/node_modules/react-native-app-auth/index.d.ts
@@ -7,14 +7,8 @@ export interface ServiceConfiguration {
 }
 
 export type BaseConfiguration =
-  | {
-      issuer?: string;
-      serviceConfiguration: ServiceConfiguration;
-    }
-  | {
-      issuer: string;
-      serviceConfiguration?: ServiceConfiguration;
-    };
+  | { serviceConfiguration: ServiceConfiguration; }
+  | { issuer: string; };
 
 type CustomHeaders = {
   authorize?: Record<string, string>;
